x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -195,6 +223,33 @@ void refresh(PersistentTasksCustomMetaData persistentTasks, ActionListener<Void>
         }
     }
 
+    void recordUpdateTimeInClusterState(ActionListener<Boolean> listener) {
+
+        clusterService.submitStateUpdateTask("ml-memory-last-update-time",
+            new AckedClusterStateUpdateTask<Boolean>(ACKED_REQUEST, listener) {
+                @Override
+                protected Boolean newResponse(boolean acknowledged) {
+                    return acknowledged;
+                }
+
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    MlMetadata currentMlMetadata = MlMetadata.getMlMetadata(currentState);
+                    MlMetadata.Builder builder = new MlMetadata.Builder(currentMlMetadata);
+                    MlMetadata newMlMetadata = builder.build();
+                    builder.setLastMemoryRefreshTime(lastUpdateTime);
```

Comment:
This line should come before `MlMetadata newMlMetadata = builder.build();` or are you wanting to compare the old and new metadatas before modifying the last refresh time? Either way line 246 should put the updated metadata

Reply:
No, it's a mistake.  I'll add a test for this too.  I was just trying to figure out how to add an end-to-end test as well.

Reply:
It's fixed and unit tested in 7ac0883d23c8b3c22979e0ddfd15abe9cc09756f.  I still need to think of how to do an end-to-end test.